qiskit/circuit/library/standard_gates/multi_control_rotation_gates.py

```diff
@@ -82,38 +83,52 @@ def _apply_mcu_graycode(circuit, theta, phi, lam, ctls, tgt, use_basis_gates):
         last_pattern = pattern
 
 
-def linear_depth_mcv(
+def mcsu2_real_diagonal(
     circuit,
-    x,
-    z,
+    unitary: np.ndarray,
     controls: Union[QuantumRegister, List[Qubit]],
-    target: Qubit,
+    target: Union[Qubit, int],
     ctrl_state: str = None,
 ):
     """
-    Apply multi-controlled SU(2) gate [[z*, x],[-x, z]] with x real and z complex.
+    Apply multi-controlled SU(2) gate with one real diagonal.
     https://arxiv.org/abs/2302.06377
 
     Args:
         circuit (QuantumCircuit): The QuantumCircuit object to apply the diagonal operator on.
-        x (float): real parameter for the single-qubit operators
-        z (complex): complex parameter for the single-qubit operators
+        unitary (ndarray): SU(2) unitary matrix with one real diagonal
         controls (QuantumRegister or list(Qubit)): The list of control qubits
-        target (Qubit): The target qubit
-        ctrl_state (str): control state of the operator SU(2) operator U
+        target (Qubit or int): The target qubit
+        ctrl_state (str): control state of the operator SU(2) operator
 
     Raises:
         QiskitError: parameter errors
     """
     # pylint: disable=cyclic-import
     from qiskit.circuit.library import MCXVChain
     from qiskit.extensions import UnitaryGate
+    from qiskit.quantum_info.operators.predicates import is_unitary_matrix
+
+    if not is_unitary_matrix(unitary):
+        raise QiskitError("parameter unitary in linear_depth_mcv must be an unitary matrix")
```

Comment:
Since the function name has changed it looks like there are 4 places which still use the old name.